### PR TITLE
feat: tauri-plugin-channel implementation

### PR DESCRIPTION
**Pull request by newfla**
PR includes the _s3/xml_uplaod progress_ refactoring based on [tauri-plugin-channel](https://github.com/fourviere/tauri-plugin-channel). 
The bug that could lead to the progress bar to be hidden has been fixed using `await sender.emit("Start");`and `let _ = receiver.started().await;`

Do **NOT** merge before [feat: TS - Split Sender and Receiver #9](https://github.com/fourviere/tauri-plugin-channel/pull/9)

### CODE REVIEW

**Comment by brainrepo** [comment]
Cool work!

File: packages/app/src-tauri/Cargo.toml [hunk 1]
```diff
@@ -41,7 +40,7 @@ tempfile = "3.8.1"
 thiserror = "1.0.56"
 tokio = { version = "1", features = ["fs", "macros"] }
 tokio-util = {version = "0.7.10", features = ["compat"] }
-uuid = {version = "1.6.1", features = ["v7", "fast-rng", "serde",] }
+tauri-plugin-channel = { git = "https://github.com/fourviere/tauri-plugin-channel.git", branch = "sender_receiver_ts" }
```
Comment (brainrepo) [inline packages/app/src-tauri/Cargo.toml:43]: Can we release the project as a package without passing it through git?

**Review by brainrepo — COMMENTED** [review]
Codewise LGTM I haven't yet chance to test it

File: packages/app/src-tauri/Cargo.toml (hunk 1 above)
Comment (newfla) [reply packages/app/src-tauri/Cargo.toml:43]: Of course, we just need to create tokens for npm and crates.io and setup the covector action on the plugin repo

**Comment by newfla** [comment]
Superseded by https://github.com/fourviere/fourviere-podcast/pull/189.

**Pull request closed without merging**